infinite/types.ts

```diff
@@ -23,17 +23,22 @@ export type SWRInfiniteKeyLoader<
   Args extends Arguments = Arguments
 > = (index: number, previousPageData: Data | null) => Args
 
+export interface SWRInfiniteCompareFn<Data = any> {
+  (a: Data | undefined, b: Data | undefined): boolean
+  (a: Data[] | undefined, b: Data[] | undefined): boolean
+}
```

Comment:
The type signature is correct, but this might be a breaking change if users put a type signature into their `compare` function.

```ts
useSWRInfinite<Data>(
  getKey,
  fetcher,
  {
    compare: (a: Data, b: Data) {
      // ...
    }
  }
);
```